noxfile.py

```diff
@@ -89,6 +88,8 @@ def default(session, install_extras=True):
         install_target = "."
     session.install("-e", install_target, "-c", constraints_path)
 
+    session.run("python", "-m", "pip", "freeze")
```

Comment:
Small comment, but would you want to continue to pip freeze in the default session going forward, or was this mostly for testing the changes in this PR?

Reply:
Theoretically, I want this in every session, cause one of the questions I ask most frequently when looking at errors is: "what version caused this situation". But I did not wanna clutter up this PR with a bunch of tweaks to all the sessions unnecessarily.